### DISCUSSION TITLE
How is spirv running with opengles on android device

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#4792] [Q&A] [answered]
Hi
  Recently, i read the code of `CodeGenerator`. I realize when `matc` output the filamat file, its content is `spirv` right? However, it seems like `Material::Builder::build` doesn't compile the `spirv` format to `glsl`. My question is how does the android device run `spirv` directly when it's backend is `opengles`.

**Comment by romainguy** [comment, marked as answer]
We use GLSL with OpenGL ES, not SPIRV-V.
